rfc/redis-removal.md

```diff
@@ -0,0 +1,171 @@
+# Redis Removal
+
+| | |
+|-|-|
+| **Status**     | Proposed |
+| **RFC #**      | [100](https://github.com/spinnaker/governance/pull/100) |
+| **Author(s)**  | Rob Zienert ([`@robzienert`](https://github.com/robzienert)) |
+| **SIG / WG**   | Platform SIG, Operations SIG |
+
+## Overview
+
+To help reduce the technical debt of the project and improve the operational
+overhead of running Spinnaker, Redis needs to be deprecated and removed from
+being a persistence option in favor of using SQL as the baseline storage
+technology. This RFC documents a process and timeline for doing so.
+
+### Goals and Non-Goals
+
+Goals:
+
+*   Provide an inventory of Spinnaker services that still need SQL support added
+*   Define a process that all services can use for migrating datastores without
+    downtime
+*   Provide a timeline for the final switch to SQL, along with criteria to
+    decide if the timeline should be delayed
+
+Non-Goals:
+
+*   Debating if SQL is the best replacement default persistence backend to
+    standardize on
+*   Documenting and solving all technical issues for each service to migrate;
+    those will be handled separately whereas this document concerns itself
+    primarily with the larger processes and timeline
+*   Identification of all new processes for maintaining services on a SQL
+    backend; these will be handled separately
+*   Prescription of what SQL vendor to use: Current functionality has been built
+    using MySQL; support for other vendors will be addressed separately
```

Comment:
Okay, but as long as the messaging makes clear that you have to switch to _MySQL_, not just _some SQL backend_ because that's the truth of the matter and the distinction is likely to be important to some customers...

I've seen people in Slack asking why they couldn't get it working with Oracle, which seems a little naïve, but I do feel like our docs are partially to blame since they refer a lot to "SQL" when I'd expect it to say MySQL if you only support the one, as we do.

Reply:
Good point, I'll update so it's clearer.